### PR TITLE
Modernize code while reading it.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/expression/fixedwidthbucketfunctionnode.cpp [hunk 1]
```diff
@@ -79,28 +78,25 @@ FixedWidthBucketFunctionNode::FloatBucketHandler::update(ResultNode &result, con
...
         _bucketHandler.reset(new FloatBucketHandler(_width->getFloat()));
     } else if (input.getClass().inherits(IntegerResultNodeVector::classId)) {
-        ResultNode::UP res(new IntegerBucketResultNodeVector());
-        setResultType(std::move(res));
+        setResultType(std::make_unique<IntegerBucketResultNodeVector>());
         _bucketHandler.reset(new IntegerVectorBucketHandler(_width->getInteger()));
     } else if (input.getClass().inherits(FloatResultNodeVector::classId)) {
         ResultNode::UP res(new FloatBucketResultNodeVector());
-        setResultType(std::move(res));
+        setResultType(std::make_unique<FloatBucketResultNodeVector>());
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/expression/fixedwidthbucketfunctionnode.cpp:99]: Consider removing  `res` above ?

File: searchlib/src/vespa/searchlib/expression/functionnode.h [hunk 2]
```diff
@@ -15,9 +14,9 @@ class FunctionNode : public ExpressionNode
     DECLARE_ABSTRACT_EXPRESSIONNODE(FunctionNode);
     const ResultNode * getResult() const override { return _tmpResult.get(); }
     ResultNode & updateResult() const { return *_tmpResult; }
-    virtual void reset() { _tmpResult.reset(NULL); }
+    virtual void reset() { _tmpResult.reset(nullptr); }
 
-    FunctionNode &setResult(const ResultNode::CP res) { _tmpResult = res; return *this; }
+    FunctionNode &setResult(const ResultNode::CP & res) { _tmpResult = res; return *this; }
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/expression/functionnode.h:19]: This change causes an extra clone() call

**Review by toregge — CHANGES_REQUESTED** [review]

File: searchlib/src/vespa/searchlib/expression/fixedwidthbucketfunctionnode.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/expression/fixedwidthbucketfunctionnode.cpp:98]: Fixed

File: searchlib/src/vespa/searchlib/expression/functionnode.h (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/expression/functionnode.h]: Fixed

**Review by toregge — APPROVED** [review]

**Pull request merged**